ledger/participant-integration-api/src/main/scala/platform/store/CompletionFromTransaction.scala

```diff
@@ -40,7 +50,17 @@ private[platform] object CompletionFromTransaction {
   ): CompletionStreamResponse =
     CompletionStreamResponse.of(
       checkpoint = Some(toApiCheckpoint(recordTime, offset)),
-      completions = Seq(Completion.of(commandId, Some(status), RejectionTransactionId)),
+      completions = Seq(
+        Completion.of(
+          commandId = commandId,
+          status = Some(status),
+          transactionId = RejectionTransactionId,
+          applicationId = null,
+          actAs = null,
+          submissionId = null,
+          deduplicationPeriod = null,
```

Comment:
In this case I wouldn't use the `.of` constructor as we set a subset of the fields. Instead I would use the default constructor `Completion(
commandId = commandId,
...
)`.

Reply:
I thought we agreed that being explicit also about the fields not being set is actually better, why not using `.of` then?

Reply:
I agree with being explicit if we set all or most fields, but in this case it seems we set a minority of the fields so I find the explicit `of` to actually make things muddier.

Reply:
If not then I'm not sure if it works out with null. We would need to pass the default for each field type, eg: "" for String, Seq.empty for repeated elems, DeduplicationPeriod.Empty for the `oneOf`

Reply:
1ba2045